wrangler-core/src/main/java/io/cdap/directives/column/SetType.java

```diff
@@ -22,44 +22,69 @@
 import io.cdap.wrangler.api.Arguments;
 import io.cdap.wrangler.api.Directive;
 import io.cdap.wrangler.api.DirectiveExecutionException;
+import io.cdap.wrangler.api.DirectiveParseException;
 import io.cdap.wrangler.api.ExecutorContext;
+import io.cdap.wrangler.api.Optional;
 import io.cdap.wrangler.api.Row;
 import io.cdap.wrangler.api.annotations.Categories;
 import io.cdap.wrangler.api.lineage.Lineage;
 import io.cdap.wrangler.api.lineage.Mutation;
 import io.cdap.wrangler.api.parser.ColumnName;
 import io.cdap.wrangler.api.parser.Identifier;
+import io.cdap.wrangler.api.parser.Numeric;
+import io.cdap.wrangler.api.parser.Text;
 import io.cdap.wrangler.api.parser.TokenType;
 import io.cdap.wrangler.api.parser.UsageDefinition;
 import io.cdap.wrangler.utils.ColumnConverter;
 
+import java.math.RoundingMode;
 import java.util.List;
 
 /**
- * A Wrangler step for converting data type of a column
+ * A Wrangler step for converting data type of column
  * Accepted types are: int, short, long, double, float, string, boolean and bytes
+ * When decimal type is selected, can also specify the scale and whether to round or not
  */
 @Plugin(type = "directives")
 @Name(SetType.NAME)
 @Categories(categories = {"column"})
-@Description("Converting data type of a column.")
+@Description("Converting data type of a column. Optional arguments scale and rounding-mode " +
+  "are used only when type is decimal.")
 public final class SetType implements Directive, Lineage {
   public static final String NAME = "set-type";
+
   private String col;
   private String type;
+  private Integer scale;
+  private RoundingMode roundingMode;
+
 
   @Override
   public UsageDefinition define() {
     UsageDefinition.Builder builder = UsageDefinition.builder(NAME);
     builder.define("column", TokenType.COLUMN_NAME);
     builder.define("type", TokenType.IDENTIFIER);
+    builder.define("scale", TokenType.NUMERIC, Optional.TRUE);
+    builder.define("rounding-mode", TokenType.TEXT, Optional.TRUE);
     return builder.build();
   }
 
   @Override
-  public void initialize(Arguments args) {
+  public void initialize(Arguments args) throws DirectiveParseException {
     col = ((ColumnName) args.value("column")).value();
     type = ((Identifier) args.value("type")).value();
+    if (type.equalsIgnoreCase("decimal")) {
+      scale = args.contains("scale") ? ((Numeric) args.value("scale")).value().intValue() : null;
+      if (scale == null && args.contains("rounding-mode")) {
+        throw new DirectiveParseException("'rounding-mode' can only be specified when a 'scale' is set");
+      }
+      try {
+        roundingMode = args.contains("rounding-mode") ?
+          RoundingMode.valueOf(((Text) args.value("rounding-mode")).value()) : RoundingMode.UNNECESSARY;
```

Comment:
This is actually where we can have a default value of, say, https://docs.oracle.com/javase/7/docs/api/java/math/RoundingMode.html#HALF_EVEN. This would make it same as floating point rounding and users would be able to use "settype decimal, 2" without rounding mode with reasonal default and without errors

Reply:
Why HALF_EVEN? Initially I assumed the default rounding mode as HALF_UP but the idea is to keep the current behaviour (no rounding if the scale is different) the same right?

Reply:
According to the documentation HALF_EVEN is the one used in IEEE-754. Also it has better math properties (statistically neutral). 

Yes, we should keep the current behavior unchanged - when scale is not set, we should do as before. Now if scale is set, we are free to provide a good default for the rounding mode as this is new behavior

Reply:
But for the default rounding mode it would be great to have references on what other systems do

Reply:
Makes sense. Done, changed the default mode to HALF_EVEN.